lib/ibm/cloud/sdk_http/base_collection.rb

```diff
@@ -1,33 +1,46 @@
 # typed: true
 # frozen_string_literal: true
 
-require_relative('base_mixins/has_child')
+require_relative 'base_http_mixin'
+require_relative 'has_child'
 
 module IBM
   module Cloud
-    module SDK
+    module SDKHTTP
       # Container that encapsulates the VPC API.
-      class BaseCollection < BaseVPC
-        # This class is used as a base for collection APIs.
-        # @param parent [Object] The parent instance in the API chain.
-        # @param endpoint [string] A path from the parent to the desired endpoint. In most cases is should be 1 word.
-        # @param array_key [string] The key that the API response holds the endpoint data. When nil the endpoint will be used.
-        # @param child_class [Object] The Object to be used when instanciating the single instance for this class.
-        def initialize(parent, endpoint, array_key: nil, child_class: nil)
+      # This class is used as a base for collection APIs.
+      # @param parent [Object] The parent instance in the API chain.
+      # @param endpoint [string] A path from the parent to the desired endpoint. In most cases is should be 1 word.
+      # @param array_key [string] The key that the API response holds the endpoint data. When nil the endpoint will be used.
+      # @param child_class [Object] The Object to be used when instanciating the single instance for this class.
+      class BaseCollection
+        def self.included(base)
+          base.send :include, BaseHTTPMixin
+        end
+        include BaseHTTPMixin
```

Comment:
I know we discussed this, but I think you can probably start paring back some of the extra includes to get to a smaller set...if you have proper specs, you can just keep removing as long as specs pass.

Reply:
Yes, I haven't tested this one. It is for the BaseMixin not HTTParty. I want to get this in and saved. Then I can pair back what is necessary. The problem that we discussed yesterday was a time where I had it 'working' and then removed a bunch of stuff without committing and broke it. Which turned out for the better but I'd like to get a working check-in and refine from there.

Reply:
Removed